app/blueprints/fund_builder/routes.py

```diff
@@ -72,8 +76,15 @@ def section(round_id):
         "round_id": str(round_id),
     }
     existing_section = None
+    # TODO there must be a better way than a pile of ifs...
     if request.args.get("action") == "remove":
-        delete_section(section_id=request.args.get("section_id"), cascade=True)
+        delete_section_from_round(round_id=round_id, section_id=request.args.get("section_id"), cascade=True)
```

Comment:
Shall we add some logs statements in to the critical points that will be easy for us to track down bugs also we'll add some comments